lib/Dialect/Torch/Transforms/RecomposeComplexOps.cpp

```diff
@@ -121,6 +121,64 @@ class RecomposeSelectFill_ : public OpRewritePattern<AtenFill_TensorOp> {
     return success();
   }
 };
+
+class RecomposeUnbindListUnpack : public OpRewritePattern<PrimListUnpackOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(PrimListUnpackOp op,
+                                PatternRewriter &rewriter) const override {
+    // recompose AtenUnbindOp + PrimListUnpackOp to select.int + squeeze ops
+    auto unbind = dyn_cast<AtenUnbindIntOp>(op.getOperand().getDefiningOp());
+    if (!unbind)
+      return failure();
+    Value dim = unbind.getDim();
+    Value input = unbind.getSelf();
+    SmallVector<Value> slices;
+    for (int i = 0; i < op.getNumResults(); i++) {
+      // rewrite to slice op
+      auto resultTy = op.getResult(i).getType();
+      auto index = rewriter.create<Torch::ConstantIntOp>(
+          op->getLoc(), rewriter.getI64IntegerAttr(i));
+      auto newSelect = rewriter.create<AtenSelectIntOp>(op->getLoc(), resultTy,
+                                                        input, dim, index);
+      auto squeeze = rewriter.create<AtenSqueezeDimOp>(op->getLoc(), resultTy,
+                                                       newSelect, dim);
+      slices.push_back(squeeze);
+    }
+    rewriter.replaceOp(op, slices);
+    return success();
+  }
+};
+
+class RecomposeUnbindGetItem : public OpRewritePattern<Aten__Getitem__TOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(Aten__Getitem__TOp op,
+                                PatternRewriter &rewriter) const override {
+    // recompose AtenUnbindIntOp + __getitem__t to select.int + squeeze op
+    auto unbind = dyn_cast<AtenUnbindIntOp>(op.getList().getDefiningOp());
+    if (!unbind)
+      return failure();
+    if (isListPotentiallyMutated(op.getList()))
```

Comment:
The function `isListPotentiallyMutated` looks at the users of the passed list. Since we are trying to prevent the case of

```python
result_list = torch.unbind(...)
mutate(result_list)
first_result = torch.__getitem__(result_list, 0)
```

then the list we actually want to pass is the result of `AtenUnbindIntOp`. This check should also be performed in the `RecomposeUnbindListUnpack` pattern.